### PR TITLE
app.test.tsx changes

### PR DESCRIPTION
**Pull request by mbondyra**
## Summary

Summarize your PR. If it involves visual changes include a screenshot or gif.

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server)
- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

### CODE REVIEW

**Review by dej611 — APPROVED** [review]
Tested locally and works great.
Noticed the `Provider` error messages but I think we can work around those later on.

File: x-pack/plugins/lens/public/app_plugin/app.test.tsx [hunk 1]
```diff
@@ -20,65 +18,70 @@ import {
...
-  const navMenuItems = {
-    expectedSaveButton: { emphasize: true, testId: 'lnsApp_saveButton' },
-    expectedSaveAsButton: { emphasize: false, testId: 'lnsApp_saveButton' },
-    expectedSaveAndReturnButton: { emphasize: true, testId: 'lnsApp_saveAndReturnButton' },
-  };
-
-  function makeDefaultProps(): jest.Mocked<LensAppProps> {
-    return {
-      editorFrame: createMockFrame(),
+  let props: jest.Mocked<LensAppProps>;
```
Comment (dej611) [inline x-pack/plugins/lens/public/app_plugin/app.test.tsx:72]: One question here.
What's the benefit of having a shared variable and reset it on every test with a `beforeEach` vs generating all the mocks explicitly every time?
It is just code readability (not calling it every time) or there's more?

File: x-pack/plugins/lens/public/app_plugin/app.test.tsx [hunk 2]
```diff
@@ -20,65 +18,70 @@ import {
...
-  return {
-    ...original,
-    debounce: (fn: unknown) => fn,
-  };
-});
+const waitToLoad = async () =>
+  await act(async () => new Promise((resolve) => setTimeout(resolve, 0)));
 
-const sessionIdSubject = new Subject<string>();
+const getLensDocumentMock = (someProps?: Partial<LensDocument>) => ({
```
Comment (dej611) [inline x-pack/plugins/lens/public/app_plugin/app.test.tsx]: In the original code there's a `cloneDeep` function that wraps this to enable document manipulation within tests without worry about influencing other tests.

File: x-pack/plugins/lens/public/app_plugin/app.test.tsx (hunk 1 above)
Comment (mbondyra) [reply x-pack/plugins/lens/public/app_plugin/app.test.tsx:72]: There's way less code to read (like calling makeDefaultProps function and then modifying the properties we want and then pass it to the `renderApp`) with this pattern so I think it's just nicer :)

File: x-pack/plugins/lens/public/app_plugin/app.test.tsx (hunk 2 above)
Comment (mbondyra) [reply x-pack/plugins/lens/public/app_plugin/app.test.tsx]: reverted 👌🏼

**Pull request merged**